misc/perf_compare.py

```diff
@@ -0,0 +1,139 @@
+"""Compare performance of mypyc-compiled mypy between one or more commits/branches.
+
+Simple usage:
+
+  python misc/perf_compare.py my-branch master ...
+
+What this does:
+
+ * Create a temp clone of the mypy repo for each target commit to measure
+ * Checkout a target commit in each of the clones
+ * Compile mypyc in each of the clones *in parallel*
+ * Create another temp clone of the mypy repo as the code to check
+ * Self check with each of the compiled mypys N times
+ * Report the average runtimes and relative performance
+ * Remove the temp clones
+"""
+
+from __future__ import annotations
+
+import argparse
+import glob
+import os
+import random
+import shutil
+import statistics
+import subprocess
+import sys
+import threading
+import time
+
+
+def heading(s: str) -> None:
+    print()
+    print(f"=== {s} ===")
+    print()
+
+
+def build_mypy(target_dir: str) -> None:
+    env = os.environ.copy()
+    env["CC"] = "clang"
+    env["MYPYC_OPT_LEVEL"] = "2"
+    cmd = ["python3", "setup.py", "--use-mypyc", "build_ext", "--inplace"]
```

Comment:
`python3` doesn't work on Windows; it's better to use `sys.executable` for cross-platform compatibility

```suggestion
    cmd = [sys.executable, "setup.py", "--use-mypyc", "build_ext", "--inplace"]
```